src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -631,8 +631,8 @@ bool WriteLog<I>::retire_entries(const unsigned long int frees_per_tx) {
             it != retiring_subentries.end(); it++) {
           ceph_assert(m_log_entries.front() == *it);
           m_log_entries.pop_front();
-          if (entry->is_write_entry()) {
-            auto write_entry = static_pointer_cast<WriteLogEntry>(entry);
```

Comment:
Nice catch!  Should `entry` variable be gotten rid of altogether to avoid a similar mishap in the future?
```
uint64_t control_block_pos = m_log_entries.front()->log_entry_index;
```

Reply:
done